asm.s

```diff
@@ -5,13 +5,21 @@ __bkpt:
   bkpt
   bx lr
 
-  .section .text.__control
-  .global __control
+  .section .text.__control_r
+  .global __control_r
   .thumb_func
-__control:
+__control_r:
   mrs r0, CONTROL
   bx lr
 
+  .section .text.__control_w
+  .global __control_w
+  .thumb_func
+__control_w:
+  msr CONTROL, r0
```

Comment:
Do you think we should insert `ISB` here to ensure the write has taken effect? The architecture reference manual suggests always issuing `ISB` after a write to `CONTROL`.

Reply:
(and if so, we should also put an ISB in the inline asm in `fn write()`).

Reply:
Hmm, the Cortex-M0+ and Cortex-M4 User Guides only mention to use ISB after changing the SPSEL bit, not any of the other bits, so maybe this shouldn't be done unconditionally.